### PR TITLE
Octopus Deploy on Docker quickstart

### PR DESCRIPTION
**Pull request by AdminTurnedDevOps**
# About

This is a post about creating a Docker instance on localhost for development purposes.

# Pre-requisites

- [x] The draft is complete and this post is ready to be reviewed.
- [x] I'm familiar with the [writing for Octopus TL;DR](https://style.octopus.com/writing-at-octopus-tldr).
- [x] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [x] The images I've included follow the [rules for working with images](https://style.octopus.com/images#rules-for-screenshots).

### CODE REVIEW

**Comment by wordlee** [comment]
Hi Michael, great post. A couple of requests, can you retake the login screen shot in light mode and with the login modal centered, and can you add the code to an Octopus repo. Thanks!

**Review by wordlee — CHANGES_REQUESTED** [review]
Hi Michael, great post. A couple of requests, can you retake the login screen shot in light mode and with the login modal centered, and can you add the code to an Octopus repo. Thanks!

**Comment by AdminTurnedDevOps** [comment]
@wordlee 

Hey Lee,

Here is the code: https://github.com/OctopusSamples/OctopusDeploy-Local-Docker-Env

In terms of the screenshot - I can absolutely make it light mode, but the thing is, the container by default is created with dark mode. Because of that, what the user sees vs what is shown in the blog post won't be the same. Thoughts?

**Comment by wordlee** [comment]
> @wordlee
> 
> Hey Lee,
> 
> Here is the code: https://github.com/OctopusSamples/OctopusDeploy-Local-Docker-Env
> 
> In terms of the screenshot - I can absolutely make it light mode, but the thing is, the container by default is created with dark mode. Because of that, what the user sees vs what is shown in the blog post won't be the same. Thoughts?

Interesting. Do you know if it creates it in dark mode in response to your system settings or by default? Personally, I prefer dark mode, but we decided to standardize on light mode.

File: blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+You will now see the Octopus Deploy login page:
+
+![](images/loginpage.png)
+
+Log in using the Octopus deploy username and password specified in the `.env` configuration file.
+
+You are now successfully logged into Octopus Deploy and can start using it.
+
+The code in this post is available in [GitHub](https://github.com/AdminTurnedDevOps/OctopusDeploy-Local-Docker-Env)
```
Comment (wordlee) [inline blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md:148]:
```suggestion
The code in this post is available in [GitHub](https://github.com/OctopusSamples/OctopusDeploy-Local-Docker-Env)
```

**Review by robpearson — COMMENTED** [review]
Hi Michael, this is great post. I really enjoyed reading it. Short and sweet and useful for our customers to get up and running w/ Octopus and Docker. Few suggestions on the high level clarity. :) 

I would suggest tweaking the title and description to match the post a bit better. Actually, I think your title is perfect. ;) I would tweak the description slightly. 

Learn how to configure Octopus Deploy to run on Docker on your local computer.

I'd tweak your headlines so they're more scalable. i.e. People can grok the broad context quickly.

Intro (no headline)
## Prerequisites
## Configure your environment
## Configure Docker Compose to run Octopus Containers
## Run the Docker Compose file

I think the references to localhost could be clearer. i.e. I would say something like your local computer or your local workstation instead of localhost. I'd suggest reviewing the usages and updating them.

I'd also add a link in the intro and conclusion to Matt's announcement post (https://octopus.com/blog/introducing-linux-docker-image) which goes into further detail on how to run this on Kubernetes etc. It's a good post for people who want to dig in deeper.

**Comment by AdminTurnedDevOps** [comment]
> > @wordlee
> > Hey Lee,
> > Here is the code: https://github.com/OctopusSamples/OctopusDeploy-Local-Docker-Env
> > In terms of the screenshot - I can absolutely make it light mode, but the thing is, the container by default is created with dark mode. Because of that, what the user sees vs what is shown in the blog post won't be the same. Thoughts?
> 
> Interesting. Do you know if it creates it in dark mode in response to your system settings or by default? Personally, I prefer dark mode, but we decided to standardize on light mode.

@wordlee Hm, that's a good point. I'm going to change my settings on my Mac to light mode and give it a shot.

File: blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+- A code editor like VS Code
+
+## The environment configuration
+
+The first step in creating a local containerized Octopus Deploy environment is to set up a `.env` configuration file. The `.env` allows you to save environment variables for certain values of the Docker Compose configuration. That way, you can set variables instead of hardcoding values.
+
+1. Open up a text editor and create a new file called `.env`
+2. Add in the following code to the `.env` file.
+
+This containerized environment is supposed to be for testing/development purposes only, which is why you will see the passwords in the `.env` configuration file. This is not recommended for production-based deployments.
```
[the remark this answers is not in the document]
Comment (AdminTurnedDevOps) [reply blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md:34]: @robpearson We should chat about this. The TLDR is; there are a few things that need to be done in the product itself before the containers can be considered "production-ready". Let's chat when you have some time about this.

File: blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+SA_PASSWORD=P@ssw0rd!
+OCTOPUS_ADMIN_USERNAME=admin
+OCTOPUS_ADMIN_PASSWORD=SecreTP@assw0rd
+```
+
+After the values are added to the `.env` file, you can create the Docker Compose file.
+
+## The Docker Compose file
+
+The Docker Compose file creates a running container for the Octopus Deploy environment to run on localhost. The Compose file consists of two configurations:
```
[the remark this answers is not in the document]
Comment (AdminTurnedDevOps) [reply blog/2020-11/set-up-octopus-deploy-on-docker-in-five-minutes/index.md:46]: Done :)

**Pull request merged**